### DISCUSSION TITLE
[Extend with JS]

### DISCUSSION THREAD

**Discussion by bijsven** [pocketbase/pocketbase#4191] [Q&A] [answered]
When writing a function for custom authenication, (direct authenication token) I had a issue because when running this code:

```JS
routerAdd("GET", "/auth/:token", (c) => {
    let token =  c.pathParam("token");
    
    const record = $app.dao().findRecordById("easyAuth", token)
    const AuthRec = $app.dao().findRecordById("users", record.get('user'))

    return $apis.recordAuthResponse($app, c, AuthRec);
});
```

I got this error:

```
TypeError: could not convert function call parameter 2: could not convert 8obcvioyokfuse6 to func(*dbx.SelectQuery) error at reflect.methodValueCall (native)
```

Does anyone know how to fix this, and what I did wrong?

**Comment by ganigeorgiev** [comment]
Are you sure that the reported error is from the above hook and not from somewhere else?

**Reply by bijsven** [reply]
```{
  "id": "wkc3i4607epc8b4",
  "created": "2024-01-20 09:53:51.423Z",
  "updated": "2024-01-20 09:53:51.423Z",
  "data": {
    "auth": "guest",
    "details": "TypeError: could not convert function call parameter 2: could not convert 8obcvioyokfuse6 to func(*dbx.SelectQuery) error at reflect.methodValueCall (native)",
    "error": "Something went wrong while processing your request.",
    "execTime": 1.261416,
    "method": "GET",
    "referer": "",
    "remoteIp": "172.17.0.1",
    "status": 400,
    "type": "request",
    "url": "/auth/1gh6ljpwxmyckno",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36",
    "userIp": "[HIDDEN]"
  },
  "message": "GET /auth/8obcvioyokfuse6",
  "level": 8
}```

**Reply by ganigeorgiev** [reply]
I don't understand the above. Make sure that the your error is not caused by some other hook.

If you are still not able to identify the culprit, please create a **minimal** reproducible repo.

**Reply by ganigeorgiev** [reply]
To help you with the debugging, try to add `try/catch` over each statement to see which one fails exactly. You can also start the executable with `--dev` and see if there is more info in the console.

**Reply by bijsven** [reply]
https://github.com/bijsven/fluffy-meme

**Reply by ganigeorgiev** [reply, marked as answer]
I'm not able to reproduce it. I've just tested it locally and both `findRecordById` and `recordAuthResponse` works as expected for me.

Either your real code has more than what you've shared and/or you are not reporting it properly.

**Reply by bijsven** [reply]
I have restarted my VPS and for some reason the code wasn't updated right... Works now!
